### PR TITLE
Updated formats.go and types.go

### PR DESCRIPTION
**Pull request by mestafin**
Added additional  datapoints as per the KNX Specification, with supporting pack() and unpack() functions

### CODE REVIEW

File: LICENSE copy [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```
Comment (vapourismo) [inline LICENSE copy]: What is this?

File: knx/dpt/formats.go [hunk 2]
```diff
@@ -3,7 +3,9 @@
 
 package dpt
 
-import "errors"
+import (
+	"errors"
+)
```
Comment (vapourismo) [inline knx/dpt/formats.go]: Is this the result of a `gofmt`?

File: knx/dpt/formats.go [hunk 3]
```diff
@@ -73,3 +75,18 @@ func unpackF16(data []byte, f *float32) error {
 	*f = 0.01 * float32(m) * float32(uint(1)<<e)
 	return nil
 }
+
+
+func packInt(i uint) []byte {
```
Comment (vapourismo) [inline knx/dpt/formats.go]: This packs a U8, not an int. It should be named `packU8` and accept a `uint8`.

File: knx/dpt/formats.go [hunk 4]
```diff
@@ -73,3 +75,18 @@ func unpackF16(data []byte, f *float32) error {
...
+	data := []byte{0, 0}
+	data[1] = byte(i)
+	return data
+}
+
+func unpackInt(data []byte, i *Scaling) error {
```
Comment (vapourismo) [inline knx/dpt/formats.go]: Same thing here as above. Why does this unpack to a `Scaling`?

File: knx/dpt/types.go [hunk 5]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
...
+// String generates a string representation.
+func (sc Scaling) String() string {
+	return fmt.Sprintf("%d%%", uint(sc))
+}
+
+// ValueTemp DPT 9.001
 type ValueTemp float32
 
 // Pack the datapoint value.
-func (temp ValueTemp) Pack() []byte {
```
Comment (vapourismo) [inline knx/dpt/types.go]: This seems completely unrelated.

File: knx/dpt/types.go [hunk 6]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
...
-	return packF16(float32(temp))
+func (vt ValueTemp) Pack() []byte {
+	return packF16(float32(vt))
 }
 
 // Unpack the datapoint value from the given data.
-func (temp *ValueTemp) Unpack(data []byte) error {
```
Comment (vapourismo) [inline knx/dpt/types.go]: Same as above.

File: knx/dpt/types.go [hunk 7]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
...
-	return unpackF16(data, (*float32)(temp))
+func (vt *ValueTemp) Unpack(data []byte) error {
+	return unpackF16(data, (*float32)(vt))
 }
 
 // String generates a string representation.
-func (temp ValueTemp) String() string {
```
Comment (vapourismo) [inline knx/dpt/types.go]: Same aa above.

File: knx/dpt/types.go [hunk 8]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
 	if sw {
 		return "On"
 	}
-
 	return "Off"
 }
 
-// ValueTemp is DPT 9.001.
+// TrueFalse DPT 1.002
+type TrueFalse bool
```
Comment (vapourismo) [inline knx/dpt/types.go]: Should generalize `1.xxx` instead of duplicating switch.

File: knx/dpt/types.go [hunk 9]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
...
+// String generates a string representation.
+func (sw TrueFalse) String() string {
+	if sw {
+		return "True"
+	}
+	return "False"
+}
+
+// DPT 1.009 OpenClose
+type OpenClose bool
```
Comment (vapourismo) [inline knx/dpt/types.go]: Same as above.

File: knx/dpt/types.go [hunk 10]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
...
+// String generates a string representation.
+func (ss StartStop) String() string {
+	if ss {
+		return "Start"
+	}
+	return "Stop"
+}
+
+// DPT 5.001  (0% - 100%)
+type Scaling int
```
Comment (vapourismo) [inline knx/dpt/types.go]: Why is this an `int`?

File: knx/dpt/types.go [hunk 11]
```diff
@@ -31,24 +32,107 @@ func (sw Switch) String() string {
...
+// Pack the datapoint value.
+func (sc Scaling) Pack() []byte {
+	sc = sc * 255 / 100
+	return packInt(uint(sc))
+}
+
+// Unpack the datapoint value from the given data.
+func (sc *Scaling) Unpack(data []byte) error {
+	var d uint = (uint(data[1]) * 100) / 255
+	data[1] = byte(d)
```
Comment (vapourismo) [inline knx/dpt/types.go]: `unpackInt` gives you `data[1]`. Why do you reassign it only to pass it to `unpackInt`?

**Comment by mestafin** [comment]
Will send you a new pull request after fixes

**Comment by vapourismo** [comment]
You can force push the changes to the source of this pull request. The PR will update automatically.
